tidy3d/components/structure.py

```diff
@@ -99,14 +101,24 @@ def eps_diagonal(self, frequency: float, coords: Coords) -> Tuple[complex, compl
     @pydantic.validator("medium", always=True)
     def _check_2d_geometry(cls, val, values):
```

Comment:
I can't help but wonder if this can just be simplified a lot

```py
geom = values.get("geometry")
if geom is None:
    raise SetupError("'Structure.geometry' did not pass validation, can't check 2D medium".)

if isinstance(val, Medium2D) and not geom.is_2D:
    raise SetupError("Structure with a `Medium2D` medium needs a `2D` geometry.")

if geom.is_2D and not isinstance(val, Medium2D):
    zero_axis = geom.bounding_box.size.index(0)
    log.warning("found a geometry with < 3 dimensions not associated with a Medium2D ...")
```

with a simple property `Geometry.is_2D` that is implemented in each of the geometry subclasses?

Reply:
Alternatives: 
* add a `Geometry.is_3D -> bool` method and check `if not geom.is_3D`
* add a `Geometry.ndims -> int` and check if `geom.ndims < 3`

Reply:
I think the complicated logic is a bit necessary, since we are checking different things. On the one hand, we need to check that if the medium is a Medium2D, the geometry must be a supported type and it must be 2D. Here, _normal_2dmaterial gives a detailed error message depending on the geometry and setup. On the other hand, if the bounding box has zero size in a certain direction and the medium is not Medium2D, it is helpful to give a warning. I'll add some comments to clarify this but I'm inclined to leave the logic as-is